c10/core/Backend.h

```diff
@@ -193,6 +207,7 @@ static inline Backend backendToHIP(Backend b) {
 constexpr DeviceType kCPU = DeviceType::CPU;
 constexpr DeviceType kCUDA = DeviceType::CUDA;
 constexpr DeviceType kHIP = DeviceType::HIP;
+constexpr DeviceType kFPGA = DeviceType::FPGA;
```

Comment:
Warning: it may not be permissible for us to reuse DeviceType::FPGA, as Caffe2 already has some OpenCL-based operators using this DeviceType, in which case the new functionality here would be in conflict. Not sure what a good alternate name is though.